src/app/core/common-components/basic-autocomplete/custom-form-control.directive.ts

```diff
@@ -139,5 +149,15 @@ export abstract class CustomFormControlDirective<T>
       this.errorState = newState;
       this.stateChanges.next();
     }
+
+    if (control) {
+      if (control.hasValidator(Validators.required)) {
+        this.required = true;
+        this.stateChanges.next();
+      } else if (this.required) {
+        this.required = false;
+        this.stateChanges.next();
+      }
```

Comment:
```suggestion
      if (this.required !== coerceBooleanProperty(control.hasValidator(Validators.required))) {
        this.required = control.hasValidator(Validators.required);
      }
```
The state change notification already happens in the setter and this makes is a bit clearer, when it is set to required.  Currently the required is always updated even if the value was always true.

Reply:
nice!